src/main/java/com/wultra/security/userdatastore/service/AttachmentService.java

```diff
@@ -0,0 +1,143 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.security.userdatastore.service;
+
+import com.wultra.core.audit.base.Audit;
+import com.wultra.core.audit.base.model.AuditDetail;
+import com.wultra.security.userdatastore.converter.AttachmentConverter;
+import com.wultra.security.userdatastore.model.dto.AttachmentDto;
+import com.wultra.security.userdatastore.model.dto.PhotoDto;
+import com.wultra.security.userdatastore.model.entity.AttachmentEntity;
+import com.wultra.security.userdatastore.model.entity.DocumentEntity;
+import com.wultra.security.userdatastore.model.entity.PhotoEntity;
+import com.wultra.security.userdatastore.model.error.ResourceNotFoundException;
+import com.wultra.security.userdatastore.model.repository.AttachmentRepository;
+import com.wultra.security.userdatastore.model.repository.DocumentRepository;
+import com.wultra.security.userdatastore.model.request.AttachmentCreateRequest;
+import com.wultra.security.userdatastore.model.request.PhotoCreateRequest;
+import com.wultra.security.userdatastore.model.response.AttachmentCreateResponse;
+import com.wultra.security.userdatastore.model.response.AttachmentResponse;
+import com.wultra.security.userdatastore.model.response.PhotoCreateResponse;
+import com.wultra.security.userdatastore.model.response.PhotoResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Service for document attachments.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Service
+@Slf4j
+public class AttachmentService {
+
+    private final AttachmentRepository attachmentRepository;
+    private final DocumentRepository documentRepository;
+    private final Audit audit;
+    private final EncryptionService encryptionService;
+    private final AttachmentConverter attachmentConverter;
+
+    @Autowired
+    AttachmentService(final AttachmentRepository attachmentRepository, final DocumentRepository documentRepository, final Audit audit, final EncryptionService encryptionService, final AttachmentConverter attachmentConverter) {
```

Comment:
You may avoid this manual writing by using `@AllArgsConstructor` if you want to.